pkg/gcrane/options.go

```diff
@@ -81,3 +82,30 @@ func WithContext(ctx context.Context) Option {
 		o.crane = append(o.crane, crane.WithContext(ctx))
 	}
 }
+
+// WithKeychain is a functional option for overriding the default
+// authenticator for remote operations, using an authn.Keychain to find
+// credentials.
+//
+// By default, gcrane will use gcrane.Keychain.
+func WithKeychain(keys authn.Keychain) Option {
+	return func(o *options) {
+		// Replace the default keychain at position 0.
```

Comment:
This feels gross, and brittle. Can we iterate options until we find the one to replace instead?

Reply:
I don't think it's possible :/

We had some discussion in https://github.com/google/go-containerregistry/pull/653 but I never came up with a better solution.